src/Controls/samples/Controls.Sample/Pages/Controls/CollectionViewGalleries/ItemsSourceGenerator.cs

```diff
@@ -156,14 +156,19 @@ void GenerateMultiTestObservableCollection()
 			}
 		}
 
-		public void GenerateEmptyObservableCollectionAndAddItemsEverySecond()
+		public void GenerateEmptyObservableCollectionAndAddItemsEverySecond(bool resetBeforeAddItems)
 		{
 			if (int.TryParse(_entry.Text, out int count))
 			{
 				var items = new ObservableCollection<CollectionViewGalleryTestItem>();
 				_cv.ItemsSource = items;
 				Dispatcher.StartTimer(TimeSpan.FromSeconds(1), () =>
 				{
+					//this test a issue with events firing out of order on IOS Obs Source
+					if(resetBeforeAddItems)
+					{
+						items.Clear();
```

Comment:
@hartez  Basic when this happens it doesn't fire the CollectionChange (reset) on the other side